src/components/labelcontainer.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import err                  from '../err';
+
+class LabelContainer extends Component{
+    constructor(props){
+        super(props);
+
+        // Defining Component Requirements
+        const requiredKeys = [
+            'id',
+            'style',
+            'onClick',
+            'errorLine',
+            'lines',
+            'colors',
+            'reference'
+        ];
+        err.missingAnyKeys('this.props',this.props,requiredKeys);
+        err.missingAnyKeys('this.props.style',this.props.style,['container','labels']);
+        this.renderLabels = this.renderLabels.bind(this);
+    }
+    renderLabels(){
+        const { errorLine, lines, colors, style } = this.props;
+        let
+            labels    = new Array(lines);
+        for(var i = 0; i < lines - 1; i++) labels[i] = i + 1;
+        return labels.map( labelNumber => {
+            const color = labelNumber !== errorLine ? colors.default : 'red';
+            return (
+                <div 
+                    key   = {labelNumber}
+                    style = {{
+                        ...style.labels,
+                        color : color
+                    }}
+                >
+                    {labelNumber}
+                </div>
+            );
+        });
+    }
+    render(){
+        const { id, style, onClick, reference } = this.props;
```

Comment:
I'm not sure about passing a reference to another component. Generally, React components should stay isolated and not pass their internals around.
I think ti would be better to keep the reference to the component within this one and find a different way to update the scroll position.